### ISSUE TITLE
conver asl onnx weight to tensorrt face problem

### ISSUE THREAD

**Issue by Chenchienyuchen** [Alibaba-MIIL/ASL#74]
I try to conver onnx to tensorrt of version 8.0.1.6

face this issue 
https://github.com/onnx/onnx-tensorrt/issues/378

Do you success to convert  onnx to tensorrt ?

Thank you!

**Comment by mrT23** [comment]
maybe a newer version of tensroRT and ONNX will support this ?

anyway, your problem is that current anti-aliasing layer uses 'reflect' padding (this is what the original article suggested):
```
    def forward(self, input):
        input_pad = F.pad(input, (1, 1, 1, 1), 'reflect')
        return F.conv2d(input_pad, self.filt, stride=self.stride, padding=0, groups=input.shape[1])
```

you can use an implementation of anti-aliasing  without padding:
```
class AntiAliasDownsampleLayerD(nn.Module):
    def __init__(self, channels):
        super(AntiAliasDownsampleLayerD, self).__init__()
        self.channels = channels
        a = torch.tensor([1., 2., 1.])

        filt = (a[:, None] * a[None, :]).clone().detach()
        filt = filt / torch.sum(filt)
        self.filt = filt[None, None, :, :].repeat((self.channels, 1, 1, 1)).cuda().half()

    def forward(self, input):
        if input.type() != self.filt.type():
            self.filt = self.filt.to(device=input.device, dtype=input.dtype)
        return F.conv2d(input, self.filt, stride=2, padding=1, groups=input.shape[1])
```

notice that this change is not bit-accurate, so you will need to retrain your models. however, you should reach the same accuracy, the 'reflect' padding doesn't contribute a lot

**Comment by Chenchienyuchen** [comment]
> maybe a newer version of tensroRT and ONNX will support this ?
> 
> anyway, your problem is that current anti-aliasing layer uses 'reflect' padding (this is what the original article suggested):
> 
> ```
>     def forward(self, input):
>         input_pad = F.pad(input, (1, 1, 1, 1), 'reflect')
>         return F.conv2d(input_pad, self.filt, stride=self.stride, padding=0, groups=input.shape[1])
> ```
> 
> you can use an implementation of anti-aliasing without padding:
> 
> ```
> class AntiAliasDownsampleLayerD(nn.Module):
>     def __init__(self, channels):
>         super(AntiAliasDownsampleLayerD, self).__init__()
>         self.channels = channels
>         a = torch.tensor([1., 2., 1.])
> 
>         filt = (a[:, None] * a[None, :]).clone().detach()
>         filt = filt / torch.sum(filt)
>         self.filt = filt[None, None, :, :].repeat((self.channels, 1, 1, 1)).cuda().half()
> 
>     def forward(self, input):
>         if input.type() != self.filt.type():
>             self.filt = self.filt.to(device=input.device, dtype=input.dtype)
>         return F.conv2d(input, self.filt, stride=2, padding=1, groups=input.shape[1])
> ```
> 
> notice that this change is not bit-accurate, so you will need to retrain your models. however, you should reach the same accuracy, the 'reflect' padding doesn't contribute a lot

It work! Thank you very much.
